### PR TITLE
Implemented onError callbacks

### PR DESCRIPTION
**Pull request by Meldiron**

### CODE REVIEW

File: src/WebSocket/Server.php [hunk 1]
```diff
@@ -32,7 +42,13 @@ public function __construct(Adapter $adapter)
      */
     public function start(): void
     {
-        $this->adapter->start();
+        try {
+            $this->adapter->start();
+        } catch(Throwable $error) {
+            foreach ($this->errorCallbacks as $errorCallback) {
+                $errorCallback($error, "utopiaWebsocket.start");
```
Comment (eldadfux) [inline src/WebSocket/Server.php:49]: I feel the event name `utopiaWebsocket.start` is a bit opinionated and non-standard. Maybe we can use simpler names like `start`, `shutdown` and so on, they will still convey the information, and the namespace seems useless at this point.

Another option is to have different exception types or exceptions codes to help distinguish between errors, the only issues is that this is a new pattern that we haven't utilized yet in other Utopia libs. @TorstenDittmann WDYT?

File: src/WebSocket/Server.php [hunk 2]
```diff
@@ -117,17 +178,36 @@ public function onMessage(callable $callback): self
...
         return $this->adapter->getConnections();
     }
+
+    /**
+     * Register callback. Will be executed when error occurs.
+     * @param callable $callback
+     * @param Throwable $error
+     * @return self
+     */
+    public function onError(callable $callback): self
```
Comment (eldadfux) [inline src/WebSocket/Server.php:208]: We have so far used the `on` prefix only for the webscoket method and the Utopia one were without any prefix (ex: start, shutdown and send), maybe we should stick to `error`?

CC @TorstenDittmann

File: src/WebSocket/Server.php (hunk 1 above)
Comment (Meldiron) [reply src/WebSocket/Server.php:49]: Remoted namespace `utopiaWebsocket` for now.

File: src/WebSocket/Server.php (hunk 2 above)
Comment (Meldiron) [reply src/WebSocket/Server.php:208]: Makes sense to me, renamed to `error`

**Review by eldadfux — APPROVED** [review]

**Review by christyjacob4 — APPROVED** [review]

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**